msm/src/logup.rs

```diff
@@ -1,11 +1,133 @@
+//!
+//! ```text
+//!   n        1          n      m(ω^i)
+//!   ∑    ----------  =  ∑    ----------                    (2)
+//!  i=1   β + f(ω^i)    i=1   β + t(ω^i)
+//! ```
+//!
+//! In the codebase, the multiplicities m_i are called the "lookup counters".
+//!
+//! The protocol can be generalized to multiple "looked-up" polynomials f_1,
+//! ..., f_k (called the "lookup terms" in the codebase) and the equation (2) becomes:
+//!
+//! ```text
+//!   n    k           1          n       m(ω^i)
+//!   ∑    ∑     ------------  =  ∑    -----------           (3)
+//!  i=1  j=1    β + f_j(ω^i)    i=1    β + t(ω^i)
+//! ```
+//!
+//! which can be rewritten as:
+//! ```text
+//!   n  (  k         1             m(ω^i)    )
+//!   ∑  (  ∑   ------------   - -----------  )  = 0         (4)
+//!  i=1 ( j=1  β + f_j(ω^i)      β + t(ω^i)  )
+//!      \                                   /
+//!       -----------------------------------
+//!                "inner sums", h(ω^i)
+//! ```
+//!
+//! The equation says that if we sum/accumulate the "inner sums" over the
+//! subgroup H, we will get a zero value. Note the analogy with the
+//! "multiplicative" accumulator used in the lookup argument called
+//! ["Plookup"](https://eprint.iacr.org/2020/315.pdf).
+//!
+//! We will define an accumulator ϕ : H -> F (called the "lookup aggregation" in
+//! the codebase) which will contain the "running
+//! inner sums" which will be equal to zero to start, and when we finished
+//! accumulating, it must equal zero. Note that the initial and final values can
+//! be anything. The idea of the equation 4 is that all the values have been
+//! read and written to the accumulator the right number of times, with respect
+//! to the multiplicities m.
+//! More precisely, we will have:
+//! ```text
+//! - φ(1) = 0
+//!                                           h(ω^j)
+//!                            /----------------------------------\
+//!                           (  k         1             m(ω^j)    )
+//! - φ(ω^{j + 1}) = φ(ω^j) + (  ∑   ------------   - -----------  )
+//!                           ( i=1  β + f_i(ω^j)      β + t(ω^j)  )
+//!
+//! - φ(ω^n) = φ(1) = 0
+//! ```
+//!
+//! We will split the inner sums into chunks of size (MAX_SUPPORTED_DEGREE - 2)
+//! to avoid having a too large degree for the quotient polynomial.
+//! As a reminder, the paper ["Multivariate lookups based on logarithmic
+//! derivatives"](https://eprint.iacr.org/2022/1530.pdf) uses the sumcheck
+//! protocol to compute the partial sums (equations 16 and 17). However, we use
+//! the PlonK polynomial IOP and therefore, we will use the quotient polynomial,
+//! and the computation of the partial sums will be translated into a constraint
+//! in a new power of alpha.
+//!
+//! Note that the inner sum h(X) can be constrainted as followed:
+//! ```text
+//!         k                   k  /          k                 \
+//! h(X) *  ᴨ  (β + f_{i}(X)) = ∑  | m_{i} *  ᴨ  (β + f_{j}(X)) |     (5)
```

Comment:
Here it should be `h(X) * \prod (\beta + \phi(i)(X))`, that is the product should include the `t`. Or you can say that `t` is just a zero (or k+1)th element alternatively.

Reply:
You're right, I noticed it and I was sure someone would see it too :D. I will modify it in https://github.com/o1-labs/proof-systems/pull/2087.